lib/browser.js

```diff
@@ -23,11 +23,24 @@ function nanoHtmlCreateElement (tag, props, children) {
     delete props.namespace
   }
 
+  // If we are extending a builtin element
+  var isCustomElement = false
+  if (props.is) {
+    isCustomElement = props.is
+    delete props.is
+  }
+
   // Create the element
   if (ns) {
-    el = document.createElementNS(ns, tag)
+    if (isCustomElement) {
+      el = document.createElementNS(ns, tag, { is: isCustomElement })
+    } else {
+      el = document.createElementNS(ns, tag)
+    }
   } else if (tag === COMMENT_TAG) {
     return document.createComment(props.comment)
+  } else if (isCustomElement) {
+    el = document.createElement(tag, { is: isCustomElement })
```

Comment:
please see the second optional `options` parameter
https://developer.mozilla.org/en-US/docs/Web/API/Document/createElement#Syntax